packages/react-router-dom/index.tsx

```diff
@@ -457,6 +474,12 @@ export function RouterProvider({
       newState: RouterState,
       { unstable_viewTransitionOpts: viewTransitionOpts }
     ) => {
+      newState.fetchers.forEach((fetcher, key) => {
+        if (fetcher.data !== undefined) {
+          fetcherData.current.set(key, fetcher.data);
+        }
+      });
```

Comment:
One-time fetcher data handoff.  As soon as a router fetcher returns to `idle`, it comes through here one time and then is deleted inside the router.  We stick the data into our `fetcherData` `Map` and clean it up when `fetcherRefs` returns to zero for a given key.

Need to look into cleanup behavior for fetchers that return data but never have a `useFetcher` register to grab it...